JBPM-6178 Stunner - Docking issues

tiagodolphine opened the pull request:
**Issues fixed:**
- Update dock position during resizing
- Persist new dock position when it has changed
- Undo/Redo operations when performing docking  and parent changing 

**Observations**
- Now it is possible to Drag an intermediate event from the palette and drag it on a node to be docked in one step. The docking action has priority over containment, for instance, it you drag an event over a task it will be docked on the closer magnet, if you drag an event over a sub process it will be docked as well, then a second step is necessary if you wanna add it as a parent. But if you drag a Task or other events other than intermediate over a sub process it will be added as a parent.
- Adjust dock points to be fixed on boundary magnets
- Check the docking rules, and they are working, now all intermediate events are able to be docked on all activities (Tasks, Sub process) if necessary this can be configured to any restriction of BPMN spec.
- It will be great to test the containment operations, I mean parent changing with dock and not docked nodes, because the commands for this was changed.

Depends on PR https://github.com/kiegroup/lienzo-core/pull/8
Related JIRA https://issues.jboss.org/browse/JBPM-6178

@romartin 
@LuboTerifaj 
@hasys

LuboTerifaj commented:
Hi I found some issues or just my point of view of using it:

- When you are trying to dock event from canvas, you must place mouse pointer exactly on boundary. It would be more intuitive if it would be possible to place event to the activities, even event (its shape) touch activity (no just pointer). What do you think?

- When I am taking event from palette, blue frame on boundary is not displayed. If I am trying to dock from canvas, the frame is displayed.

- -> My suggestion is to unify behaviour of docking from canvas and palette.

- If I want to dock event to activity on specific edge, it is docked, but it can be also docked on another edge of shape, where I want to place it. It seems that it is docked to the nearest magnet. Is it intentional?

- It is not possible to use Redo operation after: 

 create event on canvas first,
 hen dock event to activity, 
 perform Undo operation.  

- When the shape size of activity is too small (default size), it is not possible to dock event to the bottom of shape. 

- if I want to place event from palette to subprocess (except reusable sub-process of course), it is not possible to place it there, because the event is automatically docked.

- When adding event from palette to activity, where is already docked another event and the edge which contains this event is chosen, new event cover already existing event. It is possible to move it to another edge. The behaviour of docking from canvas is alright.

- Relating to issue mentioned above, it is possible to dock more than 4 events to the activities, which shouldn't be possible when there are only 4 places available for docking.

- When you reduce size of activity which has docked event(s) to the smallest size as possible, it is not possible to "select" activity and do operation with it, because event covers the whole shape of activity. Only way to do that is to use undo operation.

LuboTerifaj commented:
There is also one small issue about connector points on shapes. If sequnece flow is connected to the center point, after save, close and open, sequence flow is not connected to the center but boundary point.

LuboTerifaj commented:
I found one more thing. If you create activity with boundary event on any edge except upper, you can move with activity tightly to to the upper edge of canvas. Then it is possible to move with event to this edge but activity stays on the same position, so the event is not displayed properly (it is cut). See attached picture.
![screenshot from 2018-03-02 15-44-57](https://user-images.githubusercontent.com/12122769/36904700-bb8e10e4-1e31-11e8-85b1-d7080a03651b.png)

tiagodolphine commented on kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-core-common/src/main/java/org/kie/workbench/common/stunner/core/graph/util/GraphUtils.java:
```diff
@@ -287,7 +288,24 @@ public static boolean hasDockedNodes(final Node<?, ? extends Edge> element) {
+        Objects.requireNonNull(element.getOutEdges());
+        return element.getOutEdges()
+                        .stream()
+                        .filter(edge -> (edge.getContent() instanceof Dock))
```
@romartin the problem is that  `Class::IsInstance` is not present on the GWT JRE emulation. So for client code I think it is better to use the `instance of`.

romartin replied:
good point, realized about that later too!! sorry :)

tiagodolphine commented on kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/canvas/command/CanvasUndockNodeCommand.java:
```diff
@@ -37,12 +41,18 @@ public CanvasUndockNodeCommand(final Node parent,
-                                     MutationContext.STATIC);
-        context.applyElementMutation(child,
-                                     MutationContext.STATIC);
+        if (Objects.nonNull(parent) && Objects.nonNull(child)) {
```
done

tiagodolphine commented:
Hey @romartin @LuboTerifaj , thanks for the review I'm updating the PR and follow are my answers to your comments.

----
@romartin 
_Bug -> When moving a docked node between magnets, like an intermediate, even in a task, it "jumps" too much.. it's sometimes difficult to place it in the magnet you want, it jumps continuously on small mouse moves..
Not sure why this happens, as the logic I've seen implemented doe not do that, right? It just checks the distance... well in fact the ideal solution should be using the value in IDockingAcceptor.getHotspotSize(). This value is used right one for adjusting the drag when the mouse is in a certain distance to the border, the idea is that this "distance" parameter is configurable. So same for magnets, on the adjust you can check the shape center's location, check the target magnets and consider only adjusting (by returning true/false on the dragAdjustmethod ) when the distance to the magnet is, at least, the right one specified by getHotspotSize(). Does that makes sense?_
**Fixed**

_Bug -> Once docking and undocking a couple of times, trying to undo and redo those actions still fails at some point...it's not an exception, but it shows some "bounds exceeded" message_
 **Fixed**

_Enhancement -> Docking from palette more than one event into the same task results in events overlapping ,as they are attached to the same magnet. Anyway don't think is a bug, it's easy to re-adjust and use another manget... just a comment, I would no spend time on this although you think it's easy and makes sense_
 **Fixed, try to find an empty magnet, this may avoid this behivior.**


_Bug (not due to this) -> Last issue is when zooming / panning , but here in fact everything gets broken, @hasys can you give it a try and do one of your reports?? :) When panning or zooming a few things got broken.. we should identify and report that, proobably a single blocker ticket with all issues found using mediators (zoom/pan), WDYT?_
*** As we talk on the meeting, this is related to an issue regarding zoom/pam, on a separete JIRA for this.**

About resize commands:
** I totally agree with that, but since this refactory will take some time, I IMO it is better to open a new JIRA for this improvement, WDYT?**

---
@LuboTerifaj 

_When you are trying to dock event from canvas, you must place mouse pointer exactly on boundary. It would be more intuitive if it would be possible to place event to the activities, even event (its shape) touch activity (no just pointer). What do you think?_
**I didnt get your point, but in fact this was not changed on the PR, the behavior of draging event on the boundary to dock it was already defined, is this it?**

_When I am taking event from palette, blue frame on boundary is not displayed. If I am trying to dock from canvas, the frame is displayed.
-> My suggestion is to unify behaviour of docking from canvas and palette._
**Yes, I agree with that, but as we discussed on the meeting it is a huge change on the code, so the idea is to open a JIRA as a improvement, that is not fesiable for this next release. For now this way as the "cheaper" one to get quick win.**

_If I want to dock event to activity on specific edge, it is docked, but it can be also docked on another edge of shape, where I want to place it. It seems that it is docked to the nearest magnet. Is it intentional?_

**Yes, this was intentional we are using the magnets to dock, this was discussed on the meeting but using magnets allow us to update the location during the resizing and or when changing parent location, in a "cheaper" way on code, we can improve that on a new JIRA as well.**

_It is not possible to use Redo operation after:
create event on canvas first,
hen dock event to activity,
perform Undo operation._

**Fixed**

_When the shape size of activity is too small (default size), it is not possible to dock event to the bottom of shape._
** Now we are getting the closest magnet to the shape center, when the shape is too small for now this can cause some limitations, but I think we can limit the shape smallest size, this was done in other JIRA, I think, @romartin  is that correct?**

_If I want to place event from palette to subprocess (except reusable sub-process of course), it is not possible to place it there, because the event is automatically docked._
**Yeah, this is a limitation for now, caused by the way we are implementing the feature of dock from palette, as mentioned before on the issue about showing the boundary in blue, for now if the shape accepts docking then dock will be done, otherwise it will try to add as a child.**

_When adding event from palette to activity, where is already docked another event and the edge which contains this event is chosen, new event cover already existing event. It is possible to move it to another edge. The behavior of docking from canvas is alright._
**Fixed, now try to find an empty magnet, and attach to it, this will avoid the behavior for now.**

_Relating to issue mentioned above, it is possible to dock more than 4 events to the activities, which shouldn't be possible when there are only 4 places available for docking._
**The same here, try to find an empty magnet, will avoid this, if all are ocupied than does not allow more docking.**

_When you reduce size of activity which has docked event(s) to the smallest size as possible, it is not possible to "select" activity and do operation with it, because event covers the whole shape of activity. Only way to do that is to use undo operation or undock events from activity._
**Limit the shape smallest size may solve this, as comments on the other issue.**


There is also one small issue about connector points on shapes. If sequnece flow is connected to the center point, after save, close and open, sequence flow is not connected to the center but boundary point.
**Great point, can you please open a new JIRA for this bug, it is related to connectors.**

_I found one more thing. If you create activity with boundary event on any edge except upper, you can move with activity tightly to to the upper edge of canvas. Then it is possible to move with event to this edge but activity stays on the same position, so the event is not displayed properly (it is cut). See attached picture._
**Great point, I didn't find a feasible way easily check that, I think we can open a new JIRA to work on this bug, I think this is not a blocker for now, WDYT?**

----
Thank you guys!

tiagodolphine commented on kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-core-common/src/main/java/org/kie/workbench/common/stunner/core/graph/util/GraphUtils.java:
```diff
@@ -287,7 +288,24 @@ public static boolean hasDockedNodes(final Node<?, ? extends Edge> element) {
+        return Objects.nonNull(element.getInEdges()) ?
+                element.getInEdges()
+                        .stream()
+                        .anyMatch(edge -> edge.getContent() instanceof Dock) :
```
@romartin the problem is that Class::IsInstance is not present on the GWT JRE emulation. So for client code I think it is better to use the instance of.

tiagodolphine commented:
Jenkins execute full downstream build

tiagodolphine commented:
Jenkins execute full downstream build

romartin commented:
@tiagodolphine yeah I'm agree on all your comments, let's open tickets for improvements. And yeah thanks for the improvement about looking for an empty magnet! cool! Just going to review & test again, will let you know asap

LuboTerifaj commented:
Hey @tiagodolphine 

See my comments below:

_When you are trying to dock event from canvas, you must place mouse pointer exactly on boundary. It would be more intuitive if it would be possible to place event to the activities, even event (its shape) touch activity (no just pointer). What do you think?
I didnt get your point, but in fact this was not changed on the PR, the behavior of draging event on the boundary to dock it was already defined, is this it?_

**I mean "docking" event, when intersection of event and activity is not empty.**

_Relating to issue mentioned above, it is possible to dock more than 4 events to the activities, which shouldn't be possible when there are only 4 places available for docking.
The same here, try to find an empty magnet, will avoid this, if all are ocupied than does not allow more docking._

**This is not fixed yet. It is still possible to dock more than 4 events. Relating to this I found another issue: If you try change event type using morph, event change its place. When you move event back, old place of event is still displayed.** (see comment in https://issues.jboss.org/browse/JBPM-6178)

Other issue I will report. Thank you!

LuboTerifaj commented:
Hi @tiagodolphine
I am looking forward to fix about morphing. 
Relating to number of events on activity, it would be great somehow forbid docking more than 4 events and display some notification (warning, error) even from palette. Would be it possible? Thank you!

tiagodolphine commented:
@LuboTerifaj the fix for morph is already pushed.
About the restriction for docking more than 4 events when dock from the palette I need to investigate if there are some workaround that could be made, this is on the lienzo-core.

tiagodolphine commented:
Jenkins execute full downstream build

tiagodolphine commented:
Jenkins retest this

tiagodolphine commented:
Jenkins retest this
Jenkins execute full downstream build

tiagodolphine commented:
Jenkins execute full downstream build

LuboTerifaj commented:
Hi @tiagodolphine 
Unfortunately, I found 2 new issues:

- If I dock event to an activity, change placement of event to another edge, after resize of activity, event goes back to previous edge.

- After adding 4 events to activity, afterwards I remove 1 event and resize the activity, this error is displayed in cycle (it is displayed again and again after click on any button of this dialogue window): "Unable to complete your request. The following exception occurred: Exception caught: (TypeError) : a.e.sd(...) is null."

LuboTerifaj commented:
@tiagodolphine Otherwise, morphing works well and it seems that there is no problem with maximum amount of events in the activities  any more. Thanks!

tiagodolphine commented:
Hey @LuboTerifaj  great you found these issues, on of them was a side effect of the previous changes. 
Well I have fixed that and tested, now it seems to be working properly, can you please review it again.

Thanks !

tiagodolphine commented:
Jenkins retest this

tiagodolphine commented:
Jenkins execute full downstream build

tiagodolphine commented:
Jenkins retest this

tiagodolphine commented:
Jenkins execute full downstream build

LuboTerifaj commented:
Jenkins retest this

LuboTerifaj commented:
Jenkins execute full downstream build

LuboTerifaj commented:
Hi @tiagodolphine 
I found again two issues:

- When I am resizing shape in certain way, error message again in cycle is displayed: "Unable to complete your request. The following exception occurred: Exception caught: Exception caught: Exception caught: (TypeError) : b is null."

- Second issue is about docked event. When resizing shape it changes placement of event on edge.

Please see videos at https://issues.jboss.org/browse/JBPM-6178.

Thanks!

LuboTerifaj commented:
I also get similar messages like this: 
![screenshot from 2018-03-14 15-58-12](https://user-images.githubusercontent.com/12122769/37410330-ac26cbaa-27a0-11e8-987c-d2afba0f1ef6.png)
But currently I don't know how to reproduce it.

tiagodolphine commented:
Hey @LuboTerifaj thanks for the review, I think both issues were related, I fixed that on https://github.com/kiegroup/lienzo-core/pull/8 on the resize I was recalculating the magnet to dock, now I fixed it using the magnet index.  Can you please review it again. Just need to wais for the lienzo-core build and then the build on this PR.

Thanks !!

tiagodolphine commented:
Jenkins retest this

tiagodolphine commented:
Jenkins execute full downstream build

tiagodolphine commented:
Hey @LuboTerifaj even with the downstream in red, the war files were generated, when you have a  chance to review.
Thanks !

LuboTerifaj commented:
@tiagodolphine something is wrong with business central. I get error after log in: 
![screenshot from 2018-03-16 09-23-16](https://user-images.githubusercontent.com/12122769/37518189-6de3ecb8-2914-11e8-8b54-75bd5af69f30.png)

LuboTerifaj commented:
Jenkins execute full downstream build

LuboTerifaj commented:
It has been already reported: https://issues.jboss.org/browse/RHBA-561

jhrcek commented:
@LuboTerifaj the breaking change was reverted upstream, the latest snapshot no longer has that bug.

LuboTerifaj commented:
Hi @tiagodolphine 
Unfortunately, it still doesn't work great. I found two new issues:

- When you create some activity with boundary events, then delete it and create new one on another place on canvas, after save, close and open, there are events on the canvas, which were docked in deleted activity. They should be deleted when activity is deleted.
![beforeclose01](https://user-images.githubusercontent.com/12122769/37588651-2c42f0b4-2b63-11e8-8bd7-dea13d8742cc.png)
![beforeclose02](https://user-images.githubusercontent.com/12122769/37588661-3018a198-2b63-11e8-88fe-58de17e7b36c.png)
![afteropen](https://user-images.githubusercontent.com/12122769/37588668-39d87b18-2b63-11e8-8b79-58e5f3beef41.png)

- Second issue is related to displaying of blue frame. When I dock 3 events on activity, moving them to different edges, blue frame is permanently displayed on fixed coordinates, even the activity position is changed. It also happened when I was moving, resizing and changing type of event using morphing.
![screenshot from 2018-03-19 10-09-19](https://user-images.githubusercontent.com/12122769/37588760-819b1852-2b63-11e8-9471-0e6ca22834bb.png)

LuboTerifaj commented:
@tiagodolphine Another thing is that morphing of docked events is pretty slow, but this can be also reported as enhancement, or is there any chance to make it faster? Thanks!

tiagodolphine commented:
Hey @LuboTerifaj about the seconde issue, regarding the blue frame, I was not able to reproduce, can you give the steps or a video to reproduce the issue. The first one is already fixed, but I need to update the PR.
About the morphing IMO it is better to open a new JIRA for this improvement, since maybe it is related to all morphing feature not only this one with docking.
Thanks.

tiagodolphine commented:
@LuboTerifaj  updated with the fix for deletion.

tiagodolphine commented:
Jenkins execute full downstream build

LuboTerifaj commented:
Hi @tiagodolphine 
Regarding to the blue frame, please see video: https://issues.jboss.org/secure/attachment/12432875/Screencast%20from%2003-20-2018%2010.38.51%20AM.webm
Thanks!

LuboTerifaj commented:
@tiagodolphine Relating to the first issue, it seems that it is fixed, but I found one more issue:
When creating process in sub-process and want to change boundary event from "sub-process" to process inside of it, events shapes are displayed on unexpected place on the canvas:
![screenshot from 2018-03-20 11-06-34](https://user-images.githubusercontent.com/12122769/37648683-ee726bd4-2c2f-11e8-8fa1-b44f8b9823e6.png)
![screenshot from 2018-03-20 11-06-50](https://user-images.githubusercontent.com/12122769/37648690-f4fd6224-2c2f-11e8-9a12-2f36174b2e50.png)
![screenshot from 2018-03-20 11-07-14](https://user-images.githubusercontent.com/12122769/37648698-f9af8bf8-2c2f-11e8-9784-47b2b6f3fcc1.png)

LuboTerifaj commented:
@tiagodolphine 
Relating to issue above, it is not possible move, resize or do any operation with the sub-process after boundary event is moved back to sub-process boundary. Error message is displayed: 
![screenshot from 2018-03-20 11-19-47](https://user-images.githubusercontent.com/12122769/37649124-30f62c7e-2c31-11e8-9957-c454f60e711c.png)
Note, it is displayed in loop if trying to resize it.

LuboTerifaj commented:
Hey @tiagodolphine I just would like to mention that, I found out, that strange behaviour of displaying shapes duplicates appears also when moving (trying docking) events between sub-processes.

tiagodolphine commented:
Hi @LuboTerifaj fix to the issue when docking on sub-process pushed. 
About the issue with the blue frame, even with the video I was not able to reproduce, what browser did you use on the test? 
Thanks

tiagodolphine commented:
Jenkins execute full downstream build

tiagodolphine commented:
Jenkins retest this

tiagodolphine commented:
Jenkins execute full downstream build

LuboTerifaj commented:
Hi @tiagodolphine 
Thanks for the fix, I am going to test it. 
I used Firefox 57.0.4.

LuboTerifaj commented:
Jenkins execute full downstream build.

LuboTerifaj commented:
Jenkins retest please.

LuboTerifaj commented:
Jenkins execute full downstream build

LuboTerifaj commented:
Hey @tiagodolphine
There was some open-stack incident, so build was not successfully executed. I hope that I can test it today. What about issue with blue frame? You cannot still reproduce it?

LuboTerifaj commented:
Hi @tiagodolphine 
Relating to frame issue, it is present in chrome as well.
I have tested it on Chrome 64.0.3282.186.

tiagodolphine commented:
Hey @LuboTerifaj I will try to reproduce again following your steps. But if it is not quick to fix, I will suggest to open a new JIRA for this bug, then we will be able to merge this PR since this is long and have lot of fixes, I thinks this is important to to the master ASAP.
WDYT? besides this issue, did you got any other? 
Thanks

LuboTerifaj commented:
@tiagodolphine 
If it takes a long time, of course I can open the JIra for it. 
But, unfortunately I found some similar issue which is related to issues which you already were fixing:

- If you create process like in the picture below and try to dock twice for example message event to the sub-process edge, it is not possible, that is right. But after this action when I resize a task, message event's position is no longer displayed on its edge (see second picture). 
Second part of this issue is that if you try to dock timer event from task to the sub-process, shape of this event is displayed two times. Even when I was repeating these actions, number of shapes were increasing.  
![screenshot from 2018-03-22 16-05-06](https://user-images.githubusercontent.com/12122769/37779096-d68c8514-2deb-11e8-8876-1c3e091b62b9.png)
![screenshot from 2018-03-22 16-08-26](https://user-images.githubusercontent.com/12122769/37779101-dab45162-2deb-11e8-8c78-51ec2d20c919.png)
![screenshot from 2018-03-22 16-15-17](https://user-images.githubusercontent.com/12122769/37779628-240dbc94-2ded-11e8-932c-97e0fdfa818c.png)

LuboTerifaj commented:
@tiagodolphine 
It is also possible to dock event from task boundary to sub-process boundary and occupy the same magnet (cover existing event):

1. create process in sub-process (as above)
2. move event from task outside of sub-process to canvas
3. dock that event to the sub-process edge

LuboTerifaj commented:
@tiagodolphine 
Another issue, which is not blocker I think is:

- If dock event to the activity, move to another edge and want to change type of activity using morph, an error is displayed: "Unable to complete your request. The following exception occurred: (TypeError) : this.c.c is null." Note that this happens only once for every event.

LuboTerifaj commented:
@tiagodolphine 
I found also issues with morphing of events docked in task in sub-process:

- If you dock event to the task, change type of event using morph, move it to another edge, it is not possible to move with neither sub-process nor task in this sub-process.

- If you dock an event to the left edge of task in sub-process, position of this event is changed to another edge and it is also not possible to move task and sub-process.

tiagodolphine commented:
Hey @LuboTerifaj @romartin I'm checking these last issues, about docking with a task inside a subprocess. I'll let you know when this is done.

About the issue with the blue frame, can you open a new JIRA for that @LuboTerifaj , I spoke to @romartin  and he agreed this is not a blocker for now, and it need to be fixed on lienzo.

Thanks guys.

LuboTerifaj commented:
@tiagodolphine 
I am looking forward for the fix.
I have reported the issue with frame: https://issues.jboss.org/browse/JBPM-7091
Thanks!

tiagodolphine commented:
hey @LuboTerifaj great, thanks to open the issue, I'm working on the other issue, I'll let you know when this is ready. 
Thanks.

tiagodolphine commented:
Jenkins retest this please.

LuboTerifaj commented:
Jenkins retest please.

LuboTerifaj commented:
Jenkins execute full downstream build.

LuboTerifaj commented:
Hi @tiagodolphine
It seems that it is still not fixed. 

- It is not possible to dock signal event from sub-process to the task when I have created process in sub-process like you can see in the picture:
![screenshot from 2018-03-28 09-41-11](https://user-images.githubusercontent.com/12122769/38016395-26f8ea4c-326f-11e8-8160-e0e2e4f56df6.png)

- After you try to dock event from activity in sub-process to sub-process, it is not possible to move with the event on edges of activity. 

- When I take event from task and try to dock it on sub-process edge, it covers already existing event. It happens in inverse way as well.

- It is not possible to move with event to the sub-process edge at all when you are trying to do it from right edge.

- Displaying event's shape twice on two different coordinates. It happens when you are moving and docking events between sub-process and activity edges.
![screenshot from 2018-03-28 10-26-19](https://user-images.githubusercontent.com/12122769/38017489-7f3c55ce-3272-11e8-9061-6ed2d4739d27.png)

- The behaviour of "redocking" is pretty strange. relating to issues described above, I am getting error that docking is not possible even there are available edges:
![screenshot from 2018-03-28 10-34-58](https://user-images.githubusercontent.com/12122769/38018252-a858a456-3274-11e8-9a38-0d3de3957108.png)

LuboTerifaj commented:
@tiagodolphine 
It is also not possible to "redock" events between two activities, which are in separate sub-processes:
![screenshot from 2018-03-28 10-47-13](https://user-images.githubusercontent.com/12122769/38018617-b2906d5e-3275-11e8-8aa3-35eeb02d31e0.png)

LuboTerifaj commented:
@tiagodolphine 
One more thing, as I already mentioned there is an issue with morphing. If you change event type using morph, position of event changes (either to another edge or totally somewhere else). After this operation it is not possible to dock any event on the edge where the type of event was changed.

LuboTerifaj commented:
Hey @tiagodolphine 
I am not sure whether this is related to this PR but I just realized that intermediate boundary event must not be a target for Sequence flow. If you create event on canvas first, target it with sequence flow, it is possible to dock event to the boundary, but it shouldn't be. Do you want to fix this issue in this PR or should I open Jira for that? 
Thank you!

tiagodolphine commented:
Hey @LuboTerifaj sorry for the late response,  it seems you tested the same version, without the fixes, maybe the build wasn't triggerd right. I tested running again from intellij and cannot reproduce the bugs with subprocesses, then I triggered again the full downstream after merging with the current master, download now the war and tested, and cannot reproduce the bugs. So I think somehow the build wasn't ok and you tested the same version.
Anyway, I got one bug to find a free magnet when changing dock parent, and pushed on lienzo, besides that about docking an intermediate event that is target from on a sequence flow and I fixed it as well. I got other issue when on containment of an activity inside a subprocess, for instance try to add an event as a child of a task that is inside a subprocess, and I fixed this as well event not related to docking (c/c @romartin) .
I will trigger the full downstream and after build can you please give a try again?
Thanks.

tiagodolphine commented:
Jenkins retest this

tiagodolphine commented:
Hey @LuboTerifaj ok if they are not blocker issues I agree to open new JIRAs, in this way we can merge because this PR has lot of fixes already and are too long.
Thanks again for the review !

cc @romartin
